clients/roscpp/include/ros/timer_manager.h

```diff
@@ -428,11 +428,36 @@ void TimerManager<T, D, E>::setPeriod(int32_t handle, const D& period)
   {
     return;
   }
+  
+  if(reset)
+  {
+    info->next_expected = T::now() + period;
+  }
+  
+  // If called in a callback, last_real has not been updated yet so (now - last_real) > period
+  // In this case, let next_expected be updated only in updateNext
```

Comment:
You should convert your small node into a unit test if it is not yet one. Then you can add it to the `test_roscpp` package (https://github.com/ros/ros_comm/tree/indigo-devel/test/test_roscpp/test). 

You can add the test to this PR.